producer.go

```diff
@@ -470,6 +577,84 @@ func (p *producer) handleWorkerDone(job *rivertype.JobRow) {
 	p.jobResultCh <- job
 }
 
+func (p *producer) pollForSettingChanges(ctx context.Context, lastPaused bool) {
+	ticker := time.NewTicker(p.config.QueueSettingsPollInterval)
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-ticker.C:
+			updatedQueue, err := p.fetchQueueSettings(ctx)
+			if err != nil {
+				p.Logger.ErrorContext(ctx, p.Name+": Error fetching queue settings", slog.String("err", err.Error()))
+				continue
+			}
+			shouldBePaused := (updatedQueue.PausedAt != nil)
+			if lastPaused != shouldBePaused {
+				action := jobControlActionPause
+				if !shouldBePaused {
+					action = jobControlActionResume
+				}
+				payload := &jobControlPayload{
+					Action: action,
+					Queue:  p.config.Queue,
+				}
+				p.Logger.InfoContext(ctx, p.Name+": Queue control state changed from polling",
+					slog.String("queue", p.config.Queue),
+					slog.String("action", string(action)),
+					slog.Bool("paused", shouldBePaused),
+				)
+
+				select {
+				case p.queueControlCh <- payload:
+					lastPaused = shouldBePaused
+				default:
+					p.Logger.WarnContext(ctx, p.Name+": Queue control notification dropped due to full buffer", slog.String("action", string(action)))
+				}
+			}
+		}
+	}
+}
+
+func (p *producer) fetchQueueSettings(ctx context.Context) (*rivertype.Queue, error) {
+	ctx, cancel := context.WithTimeout(ctx, 10*time.Second)
+	defer cancel()
+
+	return p.exec.QueueGet(ctx, p.config.Queue)
+}
+
+func (p *producer) reportQueueStatusLoop(ctx context.Context) {
+	// TODO(bgentry): initial randmized sleep?
```

Comment:
Can this be resolved? Maybe just use one of the cancellable sleep helpers with a small jitter even if we're not totally married to the exact timing. Feels better than leaving the TODO in.

Reply:
Yes, didn't mean to leave this in, but did mean to ask you about the best way to solve this. I'm realizing now that `producer` is just a service, I can use `s.CancellableSleepRandomBetween()`, right?

Reply:
Yep, exactly.